libs/vec/native/build.gradle

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+apply plugin: 'c'
+
+var os = org.gradle.internal.os.OperatingSystem.current()
+
+// This task is deliberately not depended upon. Run it manually if you
+// modify the native code,
+//   ./gradlew :libs:elasticsearch-vec:native:buildNativeLibrary
+// then commit the native library, e.g.
+//   git commit libs/vec/platform/darwin-aarch64/libvec.dylib
+//
+// Look at the disassemble:
+//  objdump --disassemble-symbols=_dot8s libs/vec/native/build/libs/vec/shared/libvec.dylib
+
+// gcc -shared -fpic -o libvec.so -I src/vec/headers/ src/vec/c/vec.c -O3
+
+tasks.register("buildNativeLibrary") {
+  dependsOn tasks.named("copyLib")
+}
+
+model {
+  platforms {
+    aarch64 {
+      architecture "aarch64"
+    }
+  }
+  components {
+    vec(NativeLibrarySpec) {
+      targetPlatform "aarch64"
+      binaries.withType(SharedLibraryBinarySpec) {
+        cCompiler.args "-O3"
```

Comment:
I think that we need also `cCompiler.args "-march=native"` (at least for linux)?

Reply:
done. And pinned to armv8-a.